### ISSUE TITLE
Branch Operand Error using LSTM & NER

### ISSUE THREAD

**Issue by fantypants** [elixir-nx/axon#312]
Host Setup
  Using ASDF
  Erlang/OTP 24 [erts-12.0] [source] [64-bit] [smp:32:32] [ds:32:32:10] [async-threads:1] [jit]
  Elixir 1.13.4 (compiled with Erlang/OTP 23)
  
  EXLA Setup:
`   EXLA.set_as_nx_default([:tpu, :cuda, :rocl, :host])`
   
   Model:
  ```
  
    def build_model(word_count) do
      Axon.input({nil, @sequence_length, 1}, "input_chars")
       |> Axon.lstm(256)
       |> then(fn {{new_cell, new_hidden}, out} ->
         out
       end)
      |> Axon.nx(fn t -> t[[0..-1//1, -1]] end)
      |> Axon.dropout(rate: 0.2)
      |> Axon.dense(word_count, activation: :softmax)

  end
  
  
   params =
    model
    |> Axon.Loop.trainer(:categorical_cross_entropy, Axon.Optimizers.adam(0.001))
    |> Axon.Loop.metric(:accuracy)
    |> Axon.Loop.handle(:iteration_completed, &log_metrics/1, every: 1)
    |> Axon.Loop.run(data, %{}, epochs: 1, iterations: 2, compiler: EXLA)
```
   

I've been hitting this error, regardless of the inputs used, as soon as it starts to train you get the following:

```
13:57:00.515 [info]  XLA service 0x7f83e4094020 initialized for platform Host (this does not guarantee that XLA will be used). Devices:

13:57:00.520 [info]    StreamExecutor device (0): Host, Default Version
** (RuntimeError) branch operand 0 must match the shape of the only parameter of branch computation 0: got operand: (token[], f32[128,100,1], f32[1,256], f32[1,256], f32[1,256], /*index=5*/f32[1,256], f32[256,256], f32[256,256], f32[256,256], f32[256,256], /*index=10*/f32[256], f32[256], f32[256], f32[256], f32[128,100,1], /*index=15*/u8[128,85], s64[], s64[], f32[], f32[85], /*index=20*/f32[256,85], f32[256], f32[256], f32[256], f32[256], /*index=25*/f32[256,256], f32[256,256], f32[256,256], f32[256,256], f32[1,256], /*index=30*/f32[1,256], f32[256,85], f32[85], u8[128,85]); computation: (p: (token[], f32[128,100,1], f32[1,256], f32[1,256], f32[1,256], /*index=5*/f32[1,256], f32[256,256], f32[256,256], f32[256,256], f32[256,256], /*index=10*/f32[256], f32[256], f32[256], f32[256], s64[], /*index=15*/f32[128,1,256], f32[128,1,256], f32[128,100,256], f32[128,100,1], f32[1,256], /*index=20*/f32[1,256], f32[1,256], f32[1,256], f32[256,256], f32[256,256], /*index=25*/f32[256,256], f32[256,256], f32[256], f32[256], f32[256], /*index=30*/f32[256], f32[256,85], f32[85], u8[128,85])) -> (token[], f32[]).
    (exla 0.2.2) lib/exla/op.ex:804: EXLA.Op.unwrap!/1
    (exla 0.2.2) lib/exla/op.ex:109: EXLA.Op.get_shape/1
    (exla 0.2.2) lib/exla/defn.ex:1500: EXLA.Defn.op_shape/1
    (exla 0.2.2) lib/exla/defn.ex:735: EXLA.Defn.to_operator/4
    (exla 0.2.2) lib/exla/defn.ex:498: EXLA.Defn.cached_recur_operator/4
    (exla 0.2.2) lib/exla/defn.ex:406: EXLA.Defn.recur_operator/3
    (elixir 1.13.4) lib/enum.ex:1715: Enum."-map_reduce/3-lists^mapfoldl
```
  
  
  I've encountered this using supplied examples, and own built models.

**Comment by seanmor5** [comment]
@fantypants Do you have a quick snippet too? This is caused by something in EXLA but likely because of an issue within Axon

**Comment by fantypants** [comment]
@josevalim Is it possibly returning nil during the loop? The temp fix you mentioned highlighted the cond somewhat fixing it, perhaps we're missing a side effect or nil value through that loop

**Comment by josevalim** [comment]
@fantypants it is something else, I am almost sure. I will look into a fix and see if i can backport it to v0.2.2.

**Comment by josevalim** [comment]
If you use EXLA from the v0.2 branch, it should be all good!
